src/main/java/ca/concordia/soen/ThrowsGenericVisitor.java

```diff
@@ -0,0 +1,72 @@
+package ca.concordia.soen;
+
+import org.eclipse.jdt.core.dom.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ThrowsGenericVisitor  extends ASTVisitor {
+    private final CompilationUnit compilationUnit;
+    int AntiPatternOccurrencesCount = 0;
+    List<AntiPatternOccurrence> ThrowsGenericOcurrencesList = new ArrayList<>();
+
+    public ThrowsGenericVisitor(CompilationUnit compilationUnit) {
+        this.compilationUnit = compilationUnit;
+    }
+
+    @Override
+    public boolean visit(MethodDeclaration node) {
+        List<Type> thrownExceptions = node.thrownExceptionTypes();
+        CompilationUnit cu = compilationUnit;
```

Comment:
(Not mandatory) Do you need this? Given that you already have the global CompilationUnit var? Just a small possible improvement :)

Reply:
Sorry :3 Removed!